test/unit/kubernetes-deploy/kubernetes_resource_test.rb

```diff
@@ -48,13 +69,137 @@ def test_fetch_events_returns_empty_hash_when_kubectl_results_empty
     dummy = DummyResource.new
     dummy.deploy_started_at = Time.now.utc - 10.seconds
 
-    stub_kubectl_response("get", "events", anything, resp: "", json: false)
+    dummy.kubectl.expects(:run).returns(["", "", SystemExit.new(0)])
     events = dummy.fetch_events
     assert_operator events, :empty?
   end
 
+  def test_can_override_hardcoded_timeout_via_an_annotation
+    basic_resource = DummyResource.new
+    assert_equal 300, basic_resource.timeout
+
+    customized_resource = DummyResource.new(definition_extras: build_timeout_metadata("60S"))
```

Comment:
Seems a bit overkill to the multiple formats here since we already have unit tests for the parser, but it doesn't hurt.